internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -0,0 +1,297 @@
+
+	proxyTarget := resolver.Target{URL: *r.proxyURL}
+	wcc := &wrappingClientConn{
+		parent:       r,
+		resolverType: proxyResolverType,
+	}
+	return proxyBuilder.Build(proxyTarget, wcc, opts)
+}
+
+func (r *delegatingResolver) ResolveNow(o resolver.ResolveNowOptions) {
+	if r.targetResolver != nil {
+		r.targetResolver.ResolveNow(o)
+	}
+	if r.proxyResolver != nil {
+		r.proxyResolver.ResolveNow(o)
+	}
+}
+
+func (r *delegatingResolver) Close() {
+	if r.targetResolver != nil {
+		r.targetResolver.Close()
+	}
+	if r.proxyResolver != nil {
+		r.proxyResolver.Close()
+	}
+}
+
+// generateCombinedAddressesLocked creates a list of combined addresses by
+// pairing each proxy address with every target address. For each pair, it
+// generates a new [resolver.Address] using the proxy address, and adding the
+// target address as the attribute along with user info.
+func (r *delegatingResolver) generateCombinedAddressesLocked() ([]resolver.Address, []resolver.Endpoint) {
+	var addresses []resolver.Address
+	for _, proxyAddr := range r.proxyAddrs {
+		for _, targetAddr := range r.targetAddrs {
+			newAddr := resolver.Address{Addr: proxyAddr.Addr}
+			newAddr = proxyattributes.Populate(newAddr, proxyattributes.Options{
+				User:        r.proxyURL.User,
+				ConnectAddr: targetAddr.Addr,
+			})
+			addresses = append(addresses, newAddr)
+		}
+	}
+
+	if r.targetEndpt == nil {
+		return addresses, nil
+	}
+
+	var endpoints []resolver.Endpoint
+	for _, proxyAddr := range r.proxyAddrs {
+		for _, endpt := range r.targetEndpt {
+			var addrs []resolver.Address
+			for _, targetAddr := range endpt.Addresses {
+				newAddr := resolver.Address{Addr: proxyAddr.Addr}
+				newAddr = proxyattributes.Populate(newAddr, proxyattributes.Options{
+					User:        r.proxyURL.User,
+					ConnectAddr: targetAddr.Addr,
+				})
+				addrs = append(addrs, newAddr)
+			}
+			endpoints = append(endpoints, resolver.Endpoint{Addresses: addrs})
+		}
+	}
+	return addresses, endpoints
+}
+
+// resolverType is an enum representing the type of resolver (target or proxy).
+type resolverType int
+
+const (
+	targetResolverType resolverType = iota
+	proxyResolverType
+)
+
+// wrappingClientConn serves as an intermediary between the parent ClientConn
+// and the child resolvers created here. It implements the resolver.ClientConn
+// interface and is passed in that capacity to the child resolvers.
+//
+// Its primary function is to aggregate addresses returned by the child
+// resolvers before passing them to the parent ClientConn. Any errors returned
+// by the child resolvers are propagated verbatim to the parent ClientConn.
+type wrappingClientConn struct {
+	parent       *delegatingResolver
+	resolverType resolverType // represents the type of resolver (target or proxy)
+}
+
+// UpdateState processes updates from the target or proxy resolver. It is called
+// twice: once by the target resolver and once by the proxy resolver. It logs
```

Comment:
// It is called twice: once by the target resolver and once by the proxy resolver

That statement is not factually correct since `UpdateState` could be called any number of times by the child resolvers, whenever they have new state to update. So, I would delete that line.